### ISSUE TITLE
Infinite recursion with class that subclasses two generic classes

### ISSUE THREAD

**Issue by JukkaL** [ambv/typehinting#131]
The following code looks okay, but it results in `maximum recursion depth exceeded`:

```
from typing import Iterator, Generic, TypeVar

T = TypeVar('T')

class G(Generic[T]):
    pass

class A(Iterator[int], G[str]):
    def __next__(self): pass

A()
```

**Comment by JukkaL** [comment]
This variant the same outcome:

```
from typing import Iterator, Generic, TypeVar

T = TypeVar('T')
S = TypeVar('S')

class G(Generic[T]):
    pass

class A(Iterator[T], G[S]):
    def __next__(self): pass

A()
```

**Comment by gvanrossum** [comment]
I can no longer repro this. I think one of the most recent changes fixed this, probably rev e7aa0d6af2e2a6efc2da778af036ff6eacdcd847

**Comment by JukkaL** [comment]
I verified that it doesn't crash any more.
